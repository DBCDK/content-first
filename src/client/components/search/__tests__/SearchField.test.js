import React from 'react';
import SearchField from '../SearchField.component';
import renderer from 'react-test-renderer';

describe('SearchField', () => {
  it('renders correctly with style', () => {
    const tree = renderer
      .create(
        <SearchField
          style={{margin: 10}}
          searching={false}
          onSearch={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('displays spinner when searching', () => {
    const tree = renderer
      .create(<SearchField searching={true} onSearch={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
