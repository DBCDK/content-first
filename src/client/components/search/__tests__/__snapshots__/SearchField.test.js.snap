// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField displays spinner when searching 1`] = `
<form
  className="input-group"
  onSubmit={[Function]}
  style={Object {}}
>
  <input
    className="form-control"
    placeholder="Søg titel/forfatter"
  />
  <span
    className="input-group-btn"
  >
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      <span
        className="spinner"
        style={
          Object {
            "display": "inline-block",
            "height": 12,
            "width": 12,
          }
        }
      />
    </button>
  </span>
</form>
`;

exports[`SearchField renders correctly with style 1`] = `
<form
  className="input-group"
  onSubmit={[Function]}
  style={
    Object {
      "margin": 10,
    }
  }
>
  <input
    className="form-control"
    placeholder="Søg titel/forfatter"
  />
  <span
    className="input-group-btn"
  >
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      🔍
    </button>
  </span>
</form>
`;
